aws/provider.go

```diff
@@ -337,6 +337,7 @@ func Provider() terraform.ResourceProvider {
 			"aws_autoscaling_policy":                           resourceAwsAutoscalingPolicy(),
 			"aws_autoscaling_schedule":                         resourceAwsAutoscalingSchedule(),
 			"aws_budgets_budget":                               resourceAwsBudgetsBudget(),
+			"aws_client_vpn_endpoint":                          resourceAwsClientVpnEndpoint(),
```

Comment:
I apologize I don't have time for a full review right now, but I figure its worth mentioning that we are preferring to add EC2 naming to EC2 service resources going forward to help consolidate the resource naming and in this case prevent any future issues where another service might introduce a "Client VPN". 👍 

```suggestion
			"aws_ec2_client_vpn_endpoint":                          resourceAwsEc2ClientVpnEndpoint(),
```

(along with matching function and filename updates)

Reply:
@bflad This actually another good question.  The API classifies this as an EC2 service but I see that Terraform also categorizes some VPC services separately even if they are still technically filed under EC2.  If you look at the Client VPN service in the console, you'll see this reflected by the fact that it's accessible in the VPC section.  Would it be preferable to name this `aws_vpc_client_vpn_endpoint`

Reply:
I would lean towards `aws_ec2_client_vpn_endpoint`. Between the two (`_ec2_` vs `_vpc_`) we do have precedent of only including `_vpc_` in the resource naming only with [EC2 API operations with "Vpc" in the name](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_Operations.html) (except for `aws_vpc_dhcp_options`). This also provides some potential future proofing if some of these EC2 networking related resources work outside VPCs should some other networking concepts be introduced in the API.

The important part to remember here is that the API naming is considerably more stable than the web console, which could (theoretically) change at any time with browser redirects.

Reply:
Okay, makes sense to me!  I'll go ahead and make the appropriate changes.

Reply:
@bflad This has been done.  See 9f077a7.